packages/block-library/src/archives/edit.js

```diff
@@ -14,6 +14,17 @@ import ServerSideRender from '@wordpress/server-side-render';
 export default function ArchivesEdit( { attributes, setAttributes } ) {
 	const { showLabel, showPostCounts, displayAsDropdown, type } = attributes;
 
+	// Prevent block support styles and additional CSS classes from being printed twice.
+	const serverSideRenderAttirbutes = {
+		...attributes,
+		className: undefined,
+		style: {
+			...attributes.style,
+			spacing: undefined,
+			typography: undefined,
```

Comment:
If border support is added in the future, it will be necessary to write `border: undefined` here.

Reply:
This doesn't filter out the `fontSize` or `fontFamily` attributes that are also used by the typography supports to generate CSS classnames. The result is that classes for those are still duplicated which goes against the stated intent of this PR.

As there are a few top-level attributes added by block supports and any block using `ServerSideRender` needs to remove them along with the style attribute, it seems like we might be able to address this [within `ServerSideRender` itself](https://github.com/WordPress/gutenberg/blob/d5915916abc45e6682f4bdb70888aa41e98aa395/packages/server-side-render/src/server-side-render.js#L91) rather than duplicate that logic across several blocks.

As a quick guide the following top-level attributes added by block supports come to mind;
- `fontSize`
- `fontFamily`
- `borderColor`
- `backgroundColor`
- `textColor`
- `gradient`

There are probably some others as well for things like align support etc.